src/clr-angular/accordion/stepper/stepper.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+  Component,
+  ContentChildren,
+  QueryList,
+  Optional,
+  ChangeDetectionStrategy,
+  Input,
+  SimpleChanges,
+  OnInit,
+  AfterViewInit,
+  OnDestroy,
+  OnChanges,
+} from '@angular/core';
+import { FormGroupDirective, NgForm } from '@angular/forms';
+import { startWith, filter } from 'rxjs/operators';
+import { Subscription } from 'rxjs';
+
+import { StepperService } from './providers/stepper.service';
+import { AccordionService } from '../providers/accordion.service';
+import { ClrStepperPanel } from './stepper-panel';
+
+@Component({
+  selector: 'form[clrStepper]',
+  template: `<ng-content></ng-content>`,
+  host: {
+    '[class.clr-accordion]': 'true',
+    '[class.clr-stepper-forms]': 'true',
+  },
+  providers: [StepperService, { provide: AccordionService, useExisting: StepperService }],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ClrStepper implements OnInit, OnChanges, AfterViewInit, OnDestroy {
+  @Input('clrInitialStep') initialPanel: string;
+  @ContentChildren(ClrStepperPanel, { descendants: true })
+  panels: QueryList<ClrStepperPanel>;
+  subscriptions: Subscription[] = [];
+  form: FormGroupDirective | NgForm;
+
+  constructor(
+    @Optional() private formGroup: FormGroupDirective,
+    @Optional() private ngForm: NgForm,
+    private stepperService: StepperService
+  ) {}
```

Comment:
We should throw an error if both `formGroup` and `ngForm` are undefined with a meaningful message about being required to use forms with the stepper.

Reply:
updated